docs/user-guide/ingest-data/for-observability/vector.md

```diff
@@ -60,11 +60,389 @@ The following rules are used when storing Vector metrics into GreptimeDB:
+
+For Loki protocol, `labels` will use Tag type in time series scenarios by default, please avoid using high-cardinality fields for these fields. `structured_metadata` will be stored as a complete JSON field.
+
+Note that since Vector's configuration doesn't allow setting headers, you cannot specify pipeline. If you need to use pipeline functionality, please consider using `greptimedb_logs` sink.
+
+## Writing Trace Data
+
+### Using OpenTelemetry Protocol (not recommended)
+
+Same reasons as **writing logs using OpenTelemetry**, we recommend using components like Alloy or Otlp-collector. This document is for demonstration only.
+
+#### Example
+
+```toml
+[sources.my_source_id]
+type = "opentelemetry"
+
+  [sources.my_source_id.grpc]
+  address = "0.0.0.0:4317"
+
+  [sources.my_source_id.http]
+  address = "0.0.0.0:4318"
+  headers = {}
+
+    [sources.my_source_id.http.keepalive]
+    max_connection_age_jitter_factor = 0.1
+    max_connection_age_secs = 300
+
+[transforms.remap_traces_1]
+inputs = ["my_source_id.traces"]
+type = "remap"
+source = """
+resource_spans = [{
+  "resource": {"attributes":[
+    {"key": "service.name", "value": {"string_value": .resources.service.name}},
+    {"key": "library", "value": {"string_value": .resources.library}},
+  ]},
+  "scope_spans": [{
+    "scope": {"name": "example-scope"},
+    "spans": [{
+      "span_id": .span_id,
+      "trace_id": .trace_id,
+      "name": .name,
+      "kind": .kind,
+      "start_time_unix_nano": to_unix_timestamp!(.start_time_unix_nano, unit: "nanoseconds"),
+      "end_time_unix_nano": to_unix_timestamp!(.end_time_unix_nano, unit: "nanoseconds"),
+      "attributes": [
+        { "key": "example.attr", "value": { "string_value": .attributes."example.attr" } },
+        { "key": "span.index", "value": { "int_value": .attributes."span.index" } }
+      ],
+      "events": [],
+      "links": [],
+      "parent_span_id": .parent_span_id,
+      "trace_state": .trace_state,
+      "status": {
+        "code": .status.code,
+        "message": .status.message
+      },
+      "dropped_attributes_count": .dropped_attributes_count,
+      "dropped_events_count": .dropped_events_count,
+      "dropped_links_count": .dropped_links_count,
+    }]
+  }]
+}]
+
+. = {
+  "resource_spans": resource_spans,
+}
+"""
+
+[sinks.emit_syslog]
+inputs = ["remap_traces"]
+type = "opentelemetry"
+
+[sinks.emit_syslog.protocol]
+type = "http"
+uri = "http://localhost:4000/v1/otlp/v1/traces"
+method = "post"
+
+[sinks.emit_syslog.protocol.encoding]
+codec = "protobuf"
+
+[sinks.emit_syslog.protocol.encoding.protobuf]
+message_type = "opentelemetry.proto.trace.v1.TracesData"
+desc_file = "opentelemetry.desc"
+
+[sinks.emit_syslog.protocol.framing]
+method = "bytes"
+
+[sinks.emit_syslog.protocol.request.headers]
+content-type = "application/x-protobuf"
+```
```

Comment:
This is too hacky. I suggest to remove this and wait for real otel sink support from upstream.